s3/src/main/scala/geotrellis/spark/io/s3/S3InputFormat.scala

```diff
@@ -48,51 +51,81 @@ abstract class S3InputFormat[K, V] extends InputFormat[K,V] with LazyLogging {
       else
         new DefaultAWSCredentialsProviderChain().getCredentials
 
-    val s3client = new com.amazonaws.services.s3.AmazonS3Client(credentials)
-
-    region match {
-      case Some(r) =>
-        s3client.setRegion(r)
-      case None =>
-    }
+    val s3client: S3Client = getS3Client(credentials)
+    for (r <- region) s3client.setRegion(r)
 
     logger.info(s"Listing Splits: bucket=$bucket prefix=$prefix")
-    logger.debug(s"Authenticationg with ID=${credentials.getAWSAccessKeyId}")
+    logger.debug(s"Authenticating with ID=${credentials.getAWSAccessKeyId}")
+
     val request = new ListObjectsRequest()
       .withBucketName(bucket)
       .withPrefix(prefix)
-      .withMaxKeys(maxKeys)
 
-    var listing: ObjectListing = null
-    var splits: List[S3InputSplit] = Nil
-    do {
-      listing = s3client.listObjects(request)
-      val split = new S3InputSplit()
+    def makeNewSplit =  {
+      val split = new S3InputSplit
       split.setCredentials(credentials)
       split.bucket = bucket
-      // avoid including "directories" in the input split, can cause 403 errors on GET
-      split.keys = listing.getObjectSummaries.map(_.getKey).filterNot(_ endsWith "/")
+      split
+    }
 
-      splits = split :: splits
-      request.setMarker(listing.getNextMarker)
-    } while (listing.isTruncated)
+    var splits: Vector[S3InputSplit] = Vector(makeNewSplit)
+
+    if (null == partitionCountConf) {
+      // By default attempt to make partitions the same size
+      val maxSplitBytes = if (null == partitionSizeConf) S3InputFormat.DEFAULT_PARTITION_BYTES else partitionSizeConf.toLong
+      logger.info(s"Building partitions of at most $maxSplitBytes bytes")
```

Comment:
You mentioned it in the PR comment, but perhaps this string should be changed to make clear that the split size is not guaranteed to be \leq maxSplitBytes.